intelmq/bots/collectors/shadowserverapi/collector.py

```diff
@@ -0,0 +1,146 @@
+"""
+Shadowserver Reports API Collector Bot
+
+SPDX-FileCopyrightText: 2020 Intelmq Team <intelmq-team@cert.at>
+SPDX-License-Identifier: AGPL-3.0-or-later
+"""
+from datetime import datetime, timedelta
+import hashlib
+import hmac
+
+from intelmq.lib.bot import CollectorBot
+from intelmq.lib.cache import Cache
+from intelmq.lib.utils import create_request_session
+from intelmq.lib.exceptions import MissingDependencyError
+
+try:
+    import requests
+except ImportError:
+    requests = None
+
+APIROOT = 'https://transform.shadowserver.org/api2/'
+
+
+class ShadowServerAPICollectorBot(CollectorBot):
+    """
+    Shadowserver Reports API Collector Bot
+
+    Parameters
+    ----------
+    apikey: str
+        Your API key
+    secret: str
+        Your API secret
+    country: str
+        the country (i.e. 'austria')
+    types: list
+        the types of report you want (i.e. 'scan', 'drones', 'intel', 'sandbox_connection', 'sinkhole_combined'); if no types are defined, the bot collects all the available reports
+    """
+
+    def init(self):
+        if requests is None:
+            raise MissingDependencyError("requests")
```

Comment:
That's already a dependency of IntelMQ itself. (in setup.py)

Reply:
fixed